modules/codegen/src/smithy4s/codegen/Renderer.scala

```diff
@@ -699,10 +736,13 @@ private[codegen] class Renderer(compilationUnit: CompilationUnit) { self =>
       newline,
       renderHintsVal(hints),
       newline,
-      values.map { case e @ EnumValue(value, intValue, _, _) =>
-        line"""case object ${NameRef(
-          e.name
-        )} extends $name("$value", "${e.name}", $intValue)"""
+      values.map { case e @ EnumValue(value, intValue, _, hints) =>
+        lines(
+          deprecationAnnotation(hints),
```

Comment:
It was at this moment that I realized we ignore all hints on enum values. Somehow it doesn't feel right 😬 

related... you can actually deprecate a list member. Feels useless, but the model accepts it. There are also (more) valid collection member hints on (e.g. documentation), maybe we should be generating these...

Reply:
> maybe we should be generating these

yeah. Let's do that in a later PR

Reply:
#600